### PR TITLE
[receiver/vcenter] Fixes Cluster Resource Attributes for Datastore Resource

### PR DESCRIPTION
**Pull request by StefanKurek**
**Description:** <Describe what has changed.>
Removed the `vcenter.cluster.name` resource attribute from all Datastore resources. 

**Link to tracking Issue:** <Issue number if applicable>
#32674 

**Testing:** <Describe what testing was performed and which tests were added.>
Unit/integration tests updated and tested. Local environment tested.

**Documentation:** <Describe the documentation added.>
New documentation generated based on the metadata.

### CODE REVIEW

File: receiver/vcenterreceiver/scraper.go
Comment (crobert-1) [inline receiver/vcenterreceiver/scraper.go:1]: Should we be removing the warning log message that the attribute will be removed, now that it's being removed?
```
	logger.Warn("[WARNING] `vcenter.cluster.name`: this attribute will be removed from the Datastore resource starting in release v0.101.0")
```

File: .chloggen/fix_vcenter-datastore-attributes.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Removes `vcenter.cluster.name` attribute from Datastores"
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [32674]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
```
Comment (crobert-1) [inline .chloggen/fix_vcenter-datastore-attributes.yaml:18]: Would it be valuable to add why it's being removed, or what potentially would be necessary from users here? I'm not sure if it's entirely necessary, but with any breaking change it's good to be as clear and descriptive as possible about what's needed from users.

File: .chloggen/fix_vcenter-datastore-attributes.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: vcenterreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Removes `vcenter.cluster.name` attribute from Datastores"
```
Comment (crobert-1) [inline .chloggen/fix_vcenter-datastore-attributes.yaml:10]:
```suggestion
note: "Removes `vcenter.cluster.name` attribute from `vcenter.datastore` metrics"
```
Nit: Feel free to drop or reword, just a suggestion.

File: receiver/vcenterreceiver/scraper.go
Comment (StefanKurek) [reply receiver/vcenterreceiver/scraper.go:1]: Good catch!

File: .chloggen/fix_vcenter-datastore-attributes.yaml (hunk 1 above)
Comment (StefanKurek) [reply .chloggen/fix_vcenter-datastore-attributes.yaml:18]: Sure, I can add something here

**Comment by StefanKurek** [comment]
@crobert-1 made most of your suggested changes.

**Review by crobert-1 — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

**Pull request merged**